### PR TITLE
D-CD-10248

### PR DESCRIPTION
**Pull request by sshulgan**
## Description
Instead of using regex to validate URIs, this PR adds the use of https://www.npmjs.com/package/is.uri to validate uris. 

The `is.uri` package is the only package I was able to find that would be able to flag theses URIs as invalid:
- `http://i-comms.truecommerce.net:42000AS2/F30510A3CH`
- `https://http//test.com`
- `https://test.c`

## Breaking Changes
None

## 3rd Party Dependency Changes
Added `"is.uri": "1.0.0"`

## Internal Tracking Number
[D-CD-10248](https://cleo-jira.atlassian.net/browse/CD-10248)

## PR Checklist
_All items should be done and checked before merging._
- [ ] **Title:** Provide a very brief, general summary.
- [ ] **Details:** Fill out the _Description_, _Breaking Changes_, _3rd Party Dependency Changes_, and _Internal Tracking Number_ sections. If there's nothing for a given section, write "None".
- [ ] **Labels:** Add one or more labels.
- [ ] **Run Unit Tests:** Locally run the Karma unit tests for this project before merging this PR.
- [ ] **Run Lint:** Lint the project before merging this PR.
- [ ] **Changelog:** If any code change in this PR will run in production, add an entry to the "Unreleased" section of the `CHANGELOG.md` file.
- [ ] **Update the Wiki:** Update the wiki with the changes for the JSF components.

### CODE REVIEW

File: projects/jsf-validation/src/lib/schema-validation.service.ts [hunk 1]
```diff
@@ -111,6 +111,19 @@ export class SchemaValidationService {
 
     return null;
   }
+
+  private static isUriValid(uri: string): boolean {
+    if (!URI_VALID_CHARS_REGEX.test(uri)) {
```
Comment (mdoyle023) [inline projects/jsf-validation/src/lib/schema-validation.service.ts:116]: Why is this needed?

Comment (sshulgan) [reply projects/jsf-validation/src/lib/schema-validation.service.ts:116]: The only downside to the `is.uri` package is that it allows non-ascii character in the uri. There are other validators that don't allow non-ascii characters, but they don't flag uris like `http://i-comms.truecommerce.net:42000AS2/F30510A3CH` as invalid. So I figured we will just have to check for invalid characters ourselves, since every other aspect of `is.uri` works as we want it.

**Review by mdoyle023 — APPROVED** [review]
